appsdk/sdk.go

```diff
@@ -427,17 +432,25 @@ func (sdk *AppFunctionsSDK) Initialize() error {
 	return nil
 }
 
-func (sdk *AppFunctionsSDK) initializeStoreClient() error {
-	var err error
+func (sdk *AppFunctionsSDK) initializeSecretProvider() error {
+
+	sdk.secretProvider = security.NewSecretProvider()
+	ok := sdk.secretProvider.CreateClient(sdk.LoggingClient, sdk.config)
```

Comment:
Should we be more defensive here?  ie. check whether `sdk.secretProvider` is nil or not before calling `CreateClient`? Otherwise, the execution will be panic out.

Reply:
NewSecretProvider() returns a new object. I don't think it can return a nil